controllers/horizon_controller.go

```diff
@@ -682,39 +643,41 @@ func (r *HorizonReconciler) renderMemcached(instance *horizonv1beta1.Horizon) *m
 			Namespace: instance.Namespace,
 		},
 	}
-}
 
-func validateHorizonSecret(secret *corev1.Secret) bool {
-	return len(secret.Data["horizon-secret"]) != 0
-}
-
-func getMemcachedServerList(ctx context.Context, h *helper.Helper, instance *horizonv1beta1.Horizon, memcachedName string) ([]string, error) {
-
-	// Define the label selector to filter on
-	labelSelector := map[string]string{
-		"app":            "memcached",
-		"memcached/name": memcachedName,
-	}
+	op, err := controllerutil.CreateOrPatch(ctx, h.GetClient(), memcached, func() error {
+		memcached.Labels = map[string]string{"service": horizon.ServiceName}
+		memcached.Spec.Replicas = instance.Spec.Replicas
+		return controllerutil.SetControllerReference(h.GetBeforeObject(), memcached, h.GetScheme())
+	})
```

Comment:
I don't think this actually works any more because of the Webhooks. It fails to create the instance because of the missing `containerImage`. Are you seeing the same result in your testing?

We can leave it here for now and try to get it working in the future though.

Reply:
It appears to be working in the CI [1].  The webhooks should actually be fine now that we are not bundling them all together in the OpenStack operator.  Since each operator is deployed instead as a dependency, its webhooks will remain fully intact and should trigger whenever you create its respective CRs regardless of what is creating them.

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_horizon-operator/143/pull-ci-openstack-k8s-operators-horizon-operator-main-horizon-operator-build-deploy-kuttl/1668841955791998976/artifacts/horizon-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/api-resources/memcacheds.yaml

Reply:
Ah nice. I haven't tried horizon since reinstalling with the new architecture. But that sounds good.